glean-core/src/lib.rs

```diff
@@ -34,6 +43,30 @@ use crate::util::{local_now_with_offset, sanitize_application_id};
 
 const GLEAN_SCHEMA_VERSION: u32 = 1;
 
+/// The object holding meta information about a Glean instance.
```

Comment:
Is it worth saying something like: "In the language bindings, this is usually a singleton and all metric recording goes to a single instance of this object.  In the Rust core, it is possible to make instances of this to make testing easier."  (Maybe that's a developer note, though, not a docstring...)  I just feel it's important to call out because it somewhat informs the design.